c10/core/impl/LocalDispatchKeySet.h

```diff
@@ -24,57 +27,44 @@
 namespace c10 {
 namespace impl {
 
-// POD version of LocalDispatchKeySet.  Declared here just so that
-// we can put it in the guards.
-// This struct encapsulates special handling for TLS initialization
-// in set_included()/included() API so that they reflect the truth.
-// If you want to create PODLocalDispatchKeySet with non-zero state,
-// use set_included() instead of default constructor.
-struct C10_API PODLocalDispatchKeySet {
-  uint64_t included_;
-  uint64_t excluded_;
+// Helper class to handle conversion between the TLS zero initialized
+// convention and the "normal" in memory layout.
+class C10_API LocalDispatchKeySetWrapper {
```

Comment:
The naming scheme here is a bit confusing. Ordinarily, I would expect a LocalDispatchKeySetWrapper to be something that wraps a LocalDispatchKeySet. But actually, this is a wrapper on PODLocalState which makes it look like LocalDispatchKeySet (but there is still a LocalDispatchKeySet which will copy the POD local state out).

To kick off the bikeshed, I suggest renaming the wrapper classes to something like "PODAsLocalDispatchKeySet" (or "PODLocalStateAsLocalDispatchKeySet" if you want to be super wordy about it)

Reply:
Another consideration here is that this is a non-owning pointer to the underlying POD state, but you still use this as a normal field inside other structs. It's extra important to emphasize that this struct aliases into the true POD state (that other TLS fetches will see), rather than being just another copy of the state (which is the semantics of LocalDispatchKeyState). So if you wanted a different naming scheme, you might also consider `LocalDispatchKeySetTLSHandle` -- emphasizing the fact that it is a handle to TLS state)

Reply:
I think the whole setup can be simplified/clarified by making `LocalDispatchKeySetWrapper` *always* hold a pointer to `_get_thread_local_state()` - i.e., give it nothing but a default constructor that does this initialization - and renaming the class accordingly, e.g. `LocalDispatchKeySetTLSHandle` per Ed's suggestion, or something similar. (Even `ThreadLocalDispatchKeySet`.) 
 
Making it so an instance of this class is by definition a facade on the TLS representation will really clarify usage, I think. (I couldn't find any ctor call sites that initialized with anything else, but if "untethered" use cases exist I'd argue from separation of concerns that we should have a different class for them.)

Reply:
I think this is reasonable. The reason for the ctr that takes a pointer is that at one point I was plumbing the tls pointer along with the dispatch keyset to save a TLS lookup. But it turned out not to be worth it to enlarge the signature. So yeah, at this point there's no reason for the decoupling. (Which also makes the `Handle` naming make a lot of sense.)

Thanks for the discussion; this is really great feedback and I think will simplify things a lot.